src/metabase/lib/js.cljs

```diff
@@ -1201,15 +1201,30 @@
       #js {:operator (name operator)
            :column   column})))
 
-;; TODO remove once all filter-parts are migrated to MBQL lib
 (defn ^:export is-column-metadata
   "Returns true if arg is an MLv2 column, ie. has `:lib/type :metadata/column`.
 
-  > **Code health:** Smelly. When is this called and why does the FE need to know? The values are supposed to be opaque,
-  and we should see if there's a better way to get the needed information."
+  > **Code health:** Single use. This is used in the expression editor to parse and
+  format expression clauses."
   [arg]
   (and (map? arg) (= :metadata/column (:lib/type arg))))
 
+(defn ^:export is-metric-metadata
+  "Returns true if arg is an MLv2 metric, ie. has `:lib/type :metadata/metric`.
+
+  > **Code health:** Single use. This is used in the expression editor to parse and
+  format expression clauses."
+  [arg]
+  (and (map? arg) (= :metadata/metric (:lib/type arg))))
+
+(defn ^:export is-segment-metadata
+  "Returns true if arg is an MLv2 metric, ie. has `:lib/type :metadata/segment`.
+
+  > **Code health:** Single use. This is used in the expression editor to parse and
+  format expression clauses."
+  [arg]
+  (and (map? arg) (= :metadata/segment (:lib/type arg))))
+
```

Comment:
It's conventional in Clojure to name predicates like this `metric-metadata?`; I'm not sure why `is-column-metadata` made it in with that name.

The CLJS name mangling will transform it into `_QMARK_`, so eg. `Lib.isMetricMetadata` will be calling `ML.metric_metadata_QMARK_`. See `valid-filter-for?` in `metabase-lib/column_types.ts` as an example.

I don't know how strong the consensus is here, but my take is that we should stick to the conventions of both languages. The only place the ugly mangled names are visible is the single `ML.foo` call inside the TS wrapper.